### PR TITLE
Improve WEB_DOMAIN documentation

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Explain that queries to the webfinger endpoint on the WEB_DOMAIN have to be proxied/redirected to the LOCAL_DOMAIN.

### CODE REVIEW

File: content/en/admin/config.md [hunk 1]
```diff
@@ -21,7 +21,17 @@ This is the unique identifier of your server in the network. It cannot be safely
 
 #### `WEB_DOMAIN`
 
-It is possible to run the Mastodon interface on one domain, while having the users' handles on a different domain, e.g. addressing users as `@alice@example.com` but accessing Mastodon on `mastodon.example.com`. This may be useful if your domain name is already used for a different website but you still want to use it as a Mastodon identifier because it looks better/shorter. This requires additional nginx configuration.
+It is possible to run the Mastodon interface on one domain, while having the users' handles on a different domain, e.g. addressing users as `@alice@example.com` but accessing Mastodon on `mastodon.example.com`. This may be useful if your domain name is already used for a different website but you still want to use it as a Mastodon identifier because it looks better/shorter.
+
+As for `LOCAL_DOMAIN`, it cannot be safely changed once set.
```
Comment (sabberworm) [inline content/en/admin/config.md:26]:
```suggestion
As with `LOCAL_DOMAIN`, `WEB_DOMAIN` cannot be safely changed once set.
```

**Comment by ThibG** [comment]
@Gargron ping

**Comment by isuldor** [comment]
Since I just ran through the documentation to install an instance yesterday, I'll provide a little feedback. I found it to be excellent, except that after I finished I proceeded to the next page and found this `WEB_DOMAIN` variable documented incompletely. I was very happy to stumble onto this PR while trying to make sense of what I need to do. This concisely describes the necessary nginx location which is a good improvement!

What's still not quite clear is when you're supposed to set `WEB_DOMAIN` in `.env.production`. Since I had already started Mastodon, I restarted all three mastodon services, but I noticed the `/auth/sign_up` endpoint was still showing the `@LOCAL_DOMAIN` in the username textbox. Now I'm assuming what I actually need to do is:

- drop the mastadon_production database (it's a fresh install, so that's fine)
- rerun the setup script
- after the setup script saves the configuration, but before it builds the database, append my `WEB_DOMAIN` to `.env.production`
- let the set script build the database and finish

And that may be completely wrong, but this is my virgin voyage using the Mastodon docs.

**Comment by ThibG** [comment]
@isuldor from what you are describing, you don't want to use `WEB_DOMAIN` at all, it's only used to explicitly have the domain part of the handle be different from what domains actually handles your installation. It sounds like you *don't* want that.

As for changing the values, please never change them once an instance is set up, as Mastodon (and probably other fediverse software) currently cannot handle properly a remote account that changes `acct:` URI or actor URI.

**Comment by isuldor** [comment]
Just to level set here a little bit, I'm talking about a fresh install where I haven't even permitted 443 connections yet. I've already blown it away and started over a couple times. And I haven't changed any variables, I've only appended a WEB_DOMAIN variable immediately after completing the installation. I thought that's the whole point of this PR. The `bundle exec rake mastodon:setup` script that builds the database schema never prompts for WEB_DOMAIN. 

> it's only used to explicitly have the domain part of the handle be different from what domains actually handles your installation

I've already stumbled into a case where the email confirmation/password recovery template appears to use the `WEB_DOMAIN//auth/confirmation?confirmation_token=` in the reset link. Isn't the whole point that the instance runs on LOCAL_DOMAIN? Seems like I would need yet another location directive on my WEB_DOMAIN host to redirect to the LOCAL_DOMAIN for `/auth/`.

**Comment by ThibG** [comment]
> Just to level set here a little bit, I'm talking about a fresh install where I haven't even permitted 443 connections yet. I've already blown it away and started over a couple times.

Good, that means you still can change them indeed!

> I've already stumbled into a case where the email confirmation/password recovery template appears to use the `WEB_DOMAIN/auth/confirmation?confirmation_token=...` in the reset link. Isn't the whole point that the instance runs on LOCAL_DOMAIN? Seems like I would need yet another location directive on my WEB_DOMAIN host to redirect to the LOCAL_DOMAIN for `/auth/`.

I think you may be confusing `WEB_DOMAIN` and `LOCAL_DOMAIN`.

Basically, what this is meant for is for `@user@LOCAL_DOMAIN` to be managed by `https://WEB_DOMAIN/@user`. It is normal for the auth path to be at the `WEB_DOMAIN`.

**Comment by isuldor** [comment]
Ahh, okay. Yes that was the mistake I was making, and I see the commit prior to my comment clarifies that point.

**Comment by ThibG** [comment]
cc @Gargron

**Pull request merged**